wallet-core/src/main/java/eu/europa/ec/eudi/wallet/transfer/openid4vp/responseGenerator/OpenId4VpSdJwtResponseGeneratorImpl.kt

```diff
@@ -0,0 +1,192 @@
+import eu.europa.ec.eudi.iso18013.transfer.DisclosedDocuments
+import eu.europa.ec.eudi.iso18013.transfer.DocItem
+import eu.europa.ec.eudi.iso18013.transfer.DocRequest
+import eu.europa.ec.eudi.iso18013.transfer.DocumentsResolver
+import eu.europa.ec.eudi.iso18013.transfer.ReaderAuth
+import eu.europa.ec.eudi.iso18013.transfer.RequestDocument
+import eu.europa.ec.eudi.iso18013.transfer.RequestedDocumentData
+import eu.europa.ec.eudi.iso18013.transfer.ResponseResult
+import eu.europa.ec.eudi.iso18013.transfer.readerauth.ReaderTrustStore
+import eu.europa.ec.eudi.iso18013.transfer.response.DeviceResponse
+import eu.europa.ec.eudi.iso18013.transfer.response.ResponseGenerator
+import eu.europa.ec.eudi.openid4vp.legalName
+import eu.europa.ec.eudi.sdjwt.HashAlgorithm
+import eu.europa.ec.eudi.sdjwt.JsonPointer
+import eu.europa.ec.eudi.sdjwt.JwtAndClaims
+import eu.europa.ec.eudi.sdjwt.KeyBindingSigner
+import eu.europa.ec.eudi.sdjwt.SdJwt
+import eu.europa.ec.eudi.sdjwt.SdJwtVerifier
+import eu.europa.ec.eudi.sdjwt.asJwtVerifier
+import eu.europa.ec.eudi.sdjwt.present
+import eu.europa.ec.eudi.sdjwt.serializeWithKeyBinding
+import eu.europa.ec.eudi.wallet.internal.Openid4VpX509CertificateTrust
+import eu.europa.ec.eudi.wallet.issue.openid4vci.DocumentManagerSdJwt
+import eu.europa.ec.eudi.wallet.issue.openid4vci.pem
+import eu.europa.ec.eudi.wallet.keystore.DEV_KEY_ALIAS
+import eu.europa.ec.eudi.wallet.keystore.KeyGenerator
+import eu.europa.ec.eudi.wallet.keystore.KeyGeneratorImpl
+import eu.europa.ec.eudi.wallet.transfer.openid4vp.OpenId4VpSdJwtRequest
+import kotlinx.coroutines.runBlocking
+import org.json.JSONObject
+import java.io.ByteArrayInputStream
+import java.security.cert.CertificateFactory
+import java.security.cert.X509Certificate
+import java.util.Base64
+
+class OpenId4VpSdJwtResponseGeneratorImpl(
+    private val documentsResolver: DocumentsResolver,
+) : ResponseGenerator<OpenId4VpSdJwtRequest>() {
+    private var readerTrustStore: ReaderTrustStore? = null
+    private val openid4VpX509CertificateTrust = Openid4VpX509CertificateTrust(readerTrustStore)
+
+    override fun createResponse(disclosedDocuments: DisclosedDocuments) = runBlocking {
+        val disclosedDocument = disclosedDocuments.documents.first()
+
+        val credentials = DocumentManagerSdJwt.getDocumentById(disclosedDocument.documentId)?.data
+                          ?: throw IllegalArgumentException()
+
+        val sdJwt = getSdJwtFromCredentials(credentials)
+
+        val jsonPointer = disclosedDocument.docRequest.requestItems.mapNotNull { item ->
+            JsonPointer.parse(item.elementIdentifier)
+        }.toSet()
+
+        val presentationSdJwt = sdJwt.present(jsonPointer)
+
+        val key = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            KeyGeneratorImpl.getSigningKey(KeyGenerator.SigningKeyConfig(KeyProperties.AUTH_DEVICE_CREDENTIAL, 60))
+        } else {
+            throw Exception()
```

Comment:
Maybe adjust to something like this?
`return@runBlocking ResponseResult.Failure(AndroidException("Build version to low."))`